### DISCUSSION TITLE
Struggling with a Nintendo 64 Cartridge

### DISCUSSION THREAD

**Discussion by ZBroadcast02** [sanni/cartreader#1004] [Need help]
I have recently bought a used copy of The Legend of Zelda: Ocarina of Time (PAL, Australia) so I could use the ROM for the Ship of Harkinia PC port (As it requires a PAL version to work).

When the cartridge arrived I used isopropyl alcohol and q-tips to clean the inside before inserting it into the N64 slot of the Sanni.

I turned the Sanni on and selected 'Nintendo 64 (3V)', then I selected 'Game Cartridge'. After searching the data base this was the message:

Name: THE LEGEND OF ZELDA
Serial: NZLP
Revision: 0
ROM Size: 32MB
Save Type: SRAM
CRC1: B044B569

After clicking the button and selecting 'Read ROM', the Sanni proceeded to save the ROM to 'N64/ROM/THE LEGEND OF ZELDA/26:...'

After loading for a bit, it sent this message:

CRC32... D6B143D6 -> Not found
Press Button...

When putting the SD card into my PC, there is in fact a Z64 file of the game in the files, and upon placing that file into my folder for N64 games, the Project64 emulator is able to recognize the ROM as 'The Legend of Zelda - Ocarina of Time (E) (M3) (V1.0)'. The first time I tried to start the game in the emulator it didn't work, instead giving this error message:

Unhandled zip code
RSP: Unknown     7C 1C 97 C0
Stopping emulation
WOuld you like to open the debugger?

I've tried redumping the game again, and strangely some instances did in fact load up the game, however when I tried to use it for Ship of Harkinian it called it an Unknown ROM and wouldn't start up (Which shouldn't of happened since the SoH Compatibility Checker makes it clear that all PAL versions of the game should work.

So in short, the Sanni is able to identify the game as Ocarina of Time, yet is struggling to find something in the process and the game doesn't work half the time and doesn't appear to be applicable to SoH despite being a PAL version of the game with even the emulator identifying it as one. Anyone know what issue I'm having?

**Comment by cmcm711** [comment]
This is usually an issue with cleaning your cartridge even more than you would think necessary. Try cleaning it a bit more. 

That said, since we’re talking about an N64 game, can you tell us where you obtained your OSCR? Did you build it or buy it? If you bought it preassembled, where did you purchase it from?

**Reply by ZBroadcast02** [reply]
I purchased it from this seller: https://www.ebay.com/usr/fullcircleembedded

I did forget to mention, I have successfully dumped other cartridges with it before, namely Knuckles Chaotix (Genesis), Super Mario Bros 3 (NES), and Super Mario 64 (N64).

I'll try to give it another clean and see if it works. 

If it was a repo cartridge, would the Sanni have been able to identify the copy like an official one? The cartridge itself looks pretty convincing to what how it's supposed to look, but just feel I should ask in case it does come to the worst.

**Reply by cmcm711** [reply]
Looks like they use the proper N64 slot, which is good. Likely still a cleaning issue if you’ve successfully dumped other cartridges before. 

The only way a clone cartridge can be recognized is if the data doesn’t match the official ROM, at which point you’ll receive a similar error. The OSCR only matches checksums to a database of known values from existing official dumps. It wouldn’t have any way to know if the cartridge isn’t legitimate (please someone else correct me if I’m wrong).

**Reply by PsychoFox11** [reply]
Yeah it doesn't even have a way to know if real ones are legitimate, just if they match what's known (which has been wrong before, but not for a big game like this).
Even with a repro you should get a consistent CRC32 value on every dump if the cart works and is clean.
As far as recognizing it, it is just displaying the title on the cart header by reading it, and just means it was able to read the header, which is possible on a cart that can be otherwise bad.
If it 'recognizes' it, that's when it will give it a file name instead of a number. The folder is just the name pulled from the header. Sometimes, especially with modern releases, the header can have something different, random characters, or in the case of Night Trap for GBA, they just copied Pokemon Ruby's header including the title, so most dumpers will try to dump it based on the info it has about that title, etc. Even dumping it correctly (using another dumper, not currently possible on OSCR), the dump log shows the name Pokemon Ruby as the cart title - just what's in the header.
So it either has issues as a cart or needs cleaned if you aren't getting a consistent CRC32 that works in Project64. If you do, and it doesn't match a known one, most likely a repro.

**Reply by ZBroadcast02** [reply]
I've cleaned it for a third time, using new Q-tips with the alcohol until they no longer picked up any grey stuff on them, and it's once again giving me the same message. 

Using the SoH Compatibility Checker identifies it as an Unknown ROM, however Project64 still recognizes it.
![image](https://github.com/user-attachments/assets/2672f34e-6e70-4204-8a47-b2c5487e3870)

**Reply by ZBroadcast02** [reply]
This is what the cartridge looks like. I got the Australian version due to needing a PAL copy.
![image](https://github.com/user-attachments/assets/41fb2bc0-9474-4d6f-a800-7d9e1cd717c2)

![image](https://github.com/user-attachments/assets/dd8599bf-d815-47d2-8eff-40d25d138c26)

**Reply by PsychoFox11** [reply]
Do you have a way of looking at the PCB? A nintendo style screwdriver that is?
Also, am I correct that the CRC32 is different each time? If it is the same it could be a good dump of a repro. If it is different every time, something isn't dumping right.
Anything on OSCR I use 99% alcohol and q-tips (well off-brand ones I like better) and clean till I only get white q-tips. The OSCR is MUCH pickier about dirt than real consoles.

**Reply by ZBroadcast02** [reply]
I plan on buying a Gamebit driver to open up the cart and see if it's the official Nintendo board. I can go ahead and check if the CRC32 is the same each time.

**Reply by ZBroadcast02** [reply]
Tested it and sure enough the CRC32 is different each time! I'll still get the gamebit driver, but it definitely looks like something just isn't dumping right like you said.

So would the card that goes into the console/Sanni be the only thing I have to clean, or is there anything else that could be causing the card to get stuck? I've cleaned this one about three times until white q-tips so it is kind of surprising that it hasn't worked yet.
![image](https://github.com/user-attachments/assets/d6bdbd40-4f7f-45ae-8ad7-37357f0e0c29)
![image](https://github.com/user-attachments/assets/3a69dc19-3cd7-49c0-96a2-de3de53a2bc1)
![image](https://github.com/user-attachments/assets/102de73f-39a7-4ca7-8cdf-7cc5630e2186)

**Reply by PsychoFox11** [reply]
I mean it could be anything preventing a pin from having a good connect from traces to issues inside a chip, but usually is something visible. Once you open it you can clean better with more pressure typically too. I'd recommend a good q-tip 99% alcohol scrub (or 90% is fine), and if that doesn't do it, get a soft rubber pencil eraser (not the harder plastic kind), and rub it on the contacts really well, to loosen up stuff, then do another alcohol/q-tip cleans (make sure you get all the little eraser bits). That's gotten a couple carts that were stubborn to dump for me. I really like the Ticondaroga erasers here, really flexible and soft:
https://www.amazon.com/Ticonderoga-Carnation-Erasers-Medium-38943

If the board has any dirt, dried up soda, etc, get that as well. Also look for broken traces/pins/connections in general. That's really about all the advice I got though. Any N64 game that works should dump with a consistent CRC32 on OSCR though in my experience. Only tweaking I've ever had to do is specify the right dump type or size in ones with weird headers, or newer ones with intentionally misleading headers - Zelda should be straightforward though. The US version was one of the first I dumped with my OSCR back on version 12 of the software.

**Reply by ZBroadcast02** [reply]
Alright, well, thank you so much for all the help! I'll definitely try out everything you've said.

I'll be sure to comment the final reaults!

**Reply by ZBroadcast02** [reply]
So the driver for the gamebit screws will be arriving this Saturday, so there's not much I can do until then.

I did want to ask, does the PCB look the same for all official Nintendo games regardless of region?

**Reply by PsychoFox11** [reply]
There can be small differences, and I'm not aware of it with any N64 but there's definitely NES games where JPN and US were different (content stripped out to use smaller ROM etc.).  If you search for Ocarina of Time under No-Intro's N64 database, several of the dumps for different versions have PCB pics, so you could view those to compare.

**Comment by ZBroadcast02** [comment]
Also want to note: 
I have successfully dumped other N64 carts among other systems with this Sanni before.

**Reply by Cephiros** [reply]
Additionally i can recommend trying another sd card, enabling the 3v3 fix. Or opening and really going to town on those contacts with a pink rubber eraser then more ipa. Oscr is very picky with anything on the contact.

To verify, It is set to 3v3 or has the vselect installed and enabled, correct?

**Comment by sanni** [comment]
There are several things you can try:

(1) Open up the cartridge and see if anything looks odd, maybe some lines are corroded or the battery leaked.

(2) Try a different cartridge that has the same size, 32MB in this case, if this one also fails, try the next steps.

(3) Try a different SD card, with HW5 SD card compatibility is quite high but there are still some cards that do not work very well.

(4) Format the SD card with [Rufus](https://rufus.ie/en/) as non bootable FAT32.

(5) Enable the internal pull-up resistors on the data lines to help against interference by removing // in front of line 524 and 525 in [N64.ino](https://github.com/sanni/cartreader/blob/master/Cart_Reader/N64.ino#L524)

(6) Disable fastcrc in config.h [for HW5](https://github.com/sanni/cartreader/blob/master/Cart_Reader/Config.h#L555) or for [HW3](https://github.com/sanni/cartreader/blob/master/Cart_Reader/Config.h#L564), this will first dump the ROM, then calculate the CRC checksum, by default the N64 code does both at the same time which can be quite stressful for the Mega2560

(7) Enable 3v3 fix in [config.h](https://github.com/sanni/cartreader/blob/master/Cart_Reader/Config.h#L429), this will run the Mega2560 at half speed.

**Reply by ZBroadcast02** [reply]
So... I seemed to have figured out the problem.

So the screwdriver arrived today, and I was able to open up the cartridge and verify that it is indeed an official Nintendo board, and not a repo:
![image](https://github.com/user-attachments/assets/277798e9-821b-4265-ab11-be2cd8d958a0)

After giving it a thorough clean with the alcohol and q-tips I reassembled it and tried again, yet was still met with the same screen.

However, then I got the idea to turn on the second switch that was by the power switch and now it's suddenly worked perfectly...
![image](https://github.com/user-attachments/assets/3398c548-b8da-4bd4-9e3f-c8ffe26a1422)
![image](https://github.com/user-attachments/assets/dd2f5210-0a02-4b7d-80ce-0d1d67065751)

So I'm guessing that Switch is mean to be turned on for larger sized files, otherwise it won't work?

**Reply by sanni** [reply]
I've just added a picture to the Wiki to explain the switches in case the enclosure does not have labels. But you should still label them out of convenience. You can kinda tell what is 3V and what 5V by looking at the display brightness but labels are better.

![gedswgrrgr](https://github.com/user-attachments/assets/6213a7d3-6b93-41a5-8832-f725c1da0766)

Systems that need to be switched to 3V have "(3V)" in their menu name.
If a cartridge is run at the wrong voltage it can confuse 0 and 1 and the dump can be bad.

**Comment by cmcm711** [comment]
That is the voltage switch. It’s required for N64 and GBA cartridges to be set to 3v. It’s a bit of a shame the seller didn’t communicate that to you or label the switches.

**Reply by PsychoFox11** [reply]
Ah yeah, pretty important knowledge to avoid damaging your carts (and possibly reader).  If using any adapters for systems beyond what is built in, be sure you know the voltage for that system, for example, Pokemon mini is also 3.3v. You don't want to overvolt or undervolt your carts. GBA/N64 are the only two built in currently that use 3.3v.
Also switch it to 5v (with no cart) when doing an update.
